src/OrleansSQLUtils/Storage/RelationalStorage.cs

```diff
@@ -146,11 +146,7 @@ public Task<IEnumerable<TResult>> ReadAsync<TResult>(string query, Action<IDbCom
                 throw new ArgumentNullException("selector");
```

Comment:
I agree, was just commenting that it might not even be an optimization to be done in the future. Still, @shayhatsor is the right guy to approve this. 
I saw a lot of tests are failing after this change, though